binders/kafka-binder/spring-cloud-stream-binder-kafka-core/src/main/java/org/springframework/cloud/stream/binder/kafka/utils/BindingUtils.java

```diff
@@ -0,0 +1,193 @@
+import org.springframework.context.ApplicationContext;
+import org.springframework.kafka.support.DefaultKafkaHeaderMapper;
+import org.springframework.kafka.support.KafkaHeaderMapper;
+import org.springframework.kafka.support.converter.MessageConverter;
+import org.springframework.kafka.support.converter.MessagingMessageConverter;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+import org.springframework.util.ObjectUtils;
+
+/**
+ * Binding Utilities.
+ *
+ * @author Gary Russell
+ * @since 4.0
+ *
+ */
+public final class BindingUtils {
+
+	private BindingUtils() {
+	}
+
+	public static MessageConverter getConsumerMessageConverter(
+			ExtendedConsumerProperties<KafkaConsumerProperties> extendedConsumerProperties,
+			ApplicationContext applicationContext, KafkaBinderConfigurationProperties configurationProperties) {
+
+		MessageConverter messageConverter;
+		if (extendedConsumerProperties.getExtension().getConverterBeanName() == null) {
+			MessagingMessageConverter mmc = new MessagingMessageConverter();
+			StandardHeaders standardHeaders = extendedConsumerProperties.getExtension()
+					.getStandardHeaders();
+			mmc.setGenerateMessageId(StandardHeaders.id.equals(standardHeaders)
+							|| StandardHeaders.both.equals(standardHeaders));
+			mmc.setGenerateTimestamp(
+					StandardHeaders.timestamp.equals(standardHeaders)
+							|| StandardHeaders.both.equals(standardHeaders));
+			KafkaHeaderMapper headerMapper = getHeaderMapper(applicationContext, configurationProperties);
+			if (headerMapper == null) {
+				headerMapper = new DefaultKafkaHeaderMapper();
+			}
+			mmc.setHeaderMapper(headerMapper);
+			messageConverter = mmc;
+		}
+		else {
+			try {
+				messageConverter = applicationContext.getBean(
+						extendedConsumerProperties.getExtension().getConverterBeanName(), MessageConverter.class);
+			}
+			catch (NoSuchBeanDefinitionException ex) {
+				throw new IllegalStateException(
+						"Converter bean not present in application context", ex);
+			}
+		}
+		return messageConverter;
+	}
+
+	@Nullable
+	public static KafkaHeaderMapper getHeaderMapper(ApplicationContext applicationContext,
+			KafkaBinderConfigurationProperties configurationProperties) {
+
+		KafkaHeaderMapper mapper = null;
+		if (configurationProperties.getHeaderMapperBeanName() != null) {
+			mapper = applicationContext.getBean(
+					configurationProperties.getHeaderMapperBeanName(),
+					KafkaHeaderMapper.class);
+		}
+		if (mapper == null) {
+			//First, try to see if there is a bean named headerMapper registered by other frameworks using the binder (for e.g. spring cloud sleuth)
+			try {
+				mapper = applicationContext.getBean("kafkaBinderHeaderMapper", KafkaHeaderMapper.class);
+			}
+			catch (BeansException be) {
+			}
+		}
+		return mapper;
+	}
+
+	public static Map<String, Object> createConsumerConfigs(boolean anonymous, String consumerGroup,
```

Comment:
I am sure everyone else besides me know what `anonymous` is - but it is unclear (to me) why it drives the auto.offset.reset here. I know this is an internal util so I am not asking for a bunch of javadocs, but maybe a 1 liner comment or slight change in var name would help clarify?

Reply:
Thx for the docs @garyrussell !